src/commands/LettuceMeetVisualizer.ts

```diff
@@ -0,0 +1,292 @@
+
+    super(client, {
+      name: 'lettucemeetvisualizer',
+      boardRequired: true,
+      enabled: true,
+      description: 'Visualizes lettuce meet board.',
+      category: 'Utility',
+      usage: client.settings.prefix.concat('lettucemeetvisualizer'),
+      requiredPermissions: ['SEND_MESSAGES'],
+    }, definition);
+  }
+
+  public async run(interaction: CommandInteraction): Promise<void> {
+    await super.defer(interaction);
+
+    // VpJg7 is the default board code.
+    const boardCode = interaction.options.getString('boardcode');
+    const filterBy = interaction.options.getString('filterby');
+    const timeBlock = interaction.options.getNumber('timeblock') || 60;
+
+    if (!boardCode) {
+      await interaction.editReply('Please provide a board code.');
+      return;
+    }
+
+    const regex = new RegExp(',+', 'g');
+    const parsedFilterBy = filterBy?.split(regex).map(key => key.trim());
+
+    const boardData = await this.getBoardData(boardCode);
+    let pollResponses = boardData.data.event.pollResponses;
+    if (parsedFilterBy) {
+      pollResponses = pollResponses.filter(response =>
+        parsedFilterBy.includes(response.user.name) || parsedFilterBy.includes(response.user.email));
+    }
+
+    let allPeople = pollResponses.map(response => response.user.name);
+
+    const bestTimes = await this.getBestTimesForBlock(boardData, timeBlock, parsedFilterBy);
+
+    const embedFields: EmbedFieldData[] = [];
+    bestTimes.slice(0, 5).forEach(([startTime, availabilityData]) => {
+      const date = new Date(startTime);
+      const time = date.toLocaleString('en-US', { dateStyle: 'long', timeStyle: 'short' });
+      const unavailablePeople = allPeople.filter(person => !availabilityData.peopleAvailable.includes(person));
+
+      let message = `${availabilityData.numAvailable}/${allPeople.length} people available.`;
+      if (unavailablePeople.length > 0) {
+        message += ` Unavailable: ${unavailablePeople.join(', ')}`;
+      }
+
+      embedFields.push({
+        name: `⏰ ${time}`,
+        value: message,
+      });
+    });
+
+    const embed =
+      new MessageEmbed()
+        .setTitle(`Best times for ${timeBlock} minutes`)
+        .setColor('#47d175')
+        .addFields(embedFields);
+
+    await interaction.editReply({ embeds: [embed] });
+  }
+
+  /**
+   * @param boardData board data
+   * @param timeBlock time block in minutes
+   * @param filterBy emails or names of the users to filter by
+   * @returns best times for the given time block
+   */
+  private async getBestTimesForBlock(
+    boardData: BoardData,
+    timeBlock: number,
+    filterBy?: string[],
+  ): Promise<[number, AvailabilityData][]> {
+    const availabilityMap = await this.getAvailabilityMap(boardData, filterBy);
+    const availabilityArray = Array.from(availabilityMap.entries());
+
+    const bestTimes: [number, AvailabilityData][] = availabilityArray.map(([startTime, availabilityData]) => {
```

Comment:
Instead of an array, we could return an object since it's a little more readable.